multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/UpdateServiceMetadataStep.java

```diff
@@ -32,7 +31,10 @@ protected OperationExecutionState executeOperation(ProcessContext context, Cloud
             client.updateServiceInstanceMetadata(serviceGuid, service.getV3Metadata());
             getStepLogger().debug(Messages.UPDATING_METADATA_OF_SERVICE_INSTANCE_0_DONE, service.getName());
         } catch (CloudOperationException e) {
-            processServiceUpdateFailure(context, service, e, METADATA);
+            processServiceActionFailure(context, service,
+                                        createCloudOperationException(e, MessageFormat.format(Messages.COULD_NOT_UPDATE_OPTIONAL_SERVICE,
```

Comment:
again, maybe take out the exception creation out of the method invocation, same for the ones below